pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -258,29 +256,29 @@ func (r *reconcilerBase) createOrPatchDeployment(ctx context.Context, declared *
 	if err != nil {
 		// Let the next reconciliation retry the patch operation for valid request.
 		if !apierrors.IsInvalid(err) {
-			return controllerutil.OperationResultNone, err
+			return nil, controllerutil.OperationResultNone, err
 		}
 		// The provided data is invalid (e.g. http://b/196922619), so delete and re-create the resource.
 		// This handles changes to immutable fields, like labels.
 		r.log.Error(err, "Managed object update failed, deleting and re-creating",
 			logFieldObject, dRef.String(),
 			logFieldKind, kind)
 		if err := deploymentClient.Delete(ctx, dRef.Name, metav1.DeleteOptions{}); err != nil {
-			return controllerutil.OperationResultNone, err
+			return nil, controllerutil.OperationResultNone, err
 		}
 		data, err := json.Marshal(declared)
 		if err != nil {
-			return controllerutil.OperationResultNone, fmt.Errorf("failed to marshal declared deployment object to byte array: %w", err)
+			return nil, controllerutil.OperationResultNone, fmt.Errorf("failed to marshal declared deployment object to byte array: %w", err)
 		}
-		_, err = deploymentClient.Patch(ctx, dRef.Name, types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: reconcilermanager.ManagerName, Force: &forcePatch})
+		deploymentAfterPatch, err = deploymentClient.Patch(ctx, dRef.Name, types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: reconcilermanager.ManagerName, Force: &forcePatch})
```

Comment:
Since this is an Apply and not just a Patch, and this method only deals with deployments, how about `appliedObj` instead of `deploymentAfterPatch`. It's shorter and less confusing when the apply performs a create, not just a patch.

Reply:
Sure, Done!